packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -129,29 +128,33 @@ function Tooltip(props) {
   const touchTimer = React.useRef();
 
   React.useEffect(() => {
-    warning(
-      !childrenRef.current.disabled ||
-        (childrenRef.current.disabled && title === '') ||
-        childrenRef.current.tagName.toLowerCase() !== 'button',
-      [
-        'Material-UI: you are providing a disabled `button` child to the Tooltip component.',
-        'A disabled element does not fire events.',
-        "Tooltip needs to listen to the child element's events to display the title.",
-        '',
-        'Place a `div` container on top of the element.',
-      ].join('\n'),
-    );
+    if (childNode) {
+      warning(
+        !childNode.disabled ||
+          (childNode.disabled && title === '') ||
+          childNode.tagName.toLowerCase() !== 'button',
+        [
+          'Material-UI: you are providing a disabled `button` child to the Tooltip component.',
+          'A disabled element does not fire events.',
+          "Tooltip needs to listen to the child element's events to display the title.",
+          '',
+          'Place a `div` container on top of the element.',
+        ].join('\n'),
+      );
+    }
+  }, [title, childNode]);
 
+  React.useEffect(() => {
     // Fallback to this default id when possible.
     // Use the random value for client side rendering only.
     // We can't use it server-side.
     defaultId.current = `mui-tooltip-${Math.round(Math.random() * 1e5)}`;
 
-    // Rerender with defaultId and childrenRef.
+    // Rerender with defaultId and childNode.
     if (openProp && !mountedRef.current) {
```

Comment:
Do we actually need to check if we're mounted here? Looks like it's a bit overdefensive and React should do the heavy lifting of flushing the effect if we're unmounted.

Reply:
Dibs https://github.com/mui-org/material-ui/pull/15291#discussion_r274359477.